app/src/components/mapPopup.spec.js

```diff
@@ -0,0 +1,113 @@
+import { MapPopup } from "./mapPopup";
+import { SearchResultMap } from "./searchResultMap";
+import { store } from "../store";
+
+jest.mock("../store");
+jest.mock("./searchResultMap");
+
+describe("MapPopup", () => {
+  let element;
+  let map;
+  let mapPopup;
+
+  beforeAll(() => {
+    setDocumentHtml(getMainHtml()); // eslint-disable-line no-undef
+    element = document.getElementById("map").querySelector(".map-pop-up");
+
+    // workaround for JSDOM not supporting layout
+    // see: https://github.com/testing-library/react-testing-library/issues/353#issuecomment-481248489
+    Object.defineProperty(element, "getBoundingClientRect", {
+      configurable: true,
+      value: () => ({ width: 250, height: 75 }),
+    });
+  });
+
+  beforeEach(() => {
+    store.getState.mockImplementation(() => ({
+      addressGeocode: {
+        searchResult: null,
+        status: "idle",
+        error: null,
+      },
+      rentStabilized: {
+        status: "idle",
+        error: null,
+        match: null,
+      },
+    }));
+
+    fetch.mockResponse(
+      JSON.stringify({
+        layergroupid: "blablabla",
+        cdn_url: {
+          templates: {
+            https: {
+              url: "https://cartocdn-gusc-{s}.global.ssl.fastly.net",
+              subdomains: ["a", "b", "c", "d"],
+            },
+          },
+        },
+      }),
+      { status: 200, statusText: "OK" }
+    );
+
+    map = new SearchResultMap({
+      element: document.getElementById("map"),
+      store,
+    });
+    mapPopup = new MapPopup({ element, map });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    jest.resetModules();
+  });
+
+  test("The consumer should be able to call new() on MapPopup", () => {
+    expect(mapPopup).toBeTruthy();
+  });
+
+  test("The component's HTML exists", () => {
+    expect(mapPopup.element).toBeDefined();
```

Comment:
should maybe change this test from using `toBeDefined()` to something more strict? And maybe update all tests?